### ISSUE TITLE
File uploads/downloads, clicks on links not working on Ubuntu

### ISSUE THREAD

**Issue by andreyeltsin** [telegramdesktop/tdesktop#7701]
### Steps to reproduce
1. Open Telegram
2. Try to upload/download any file/click on link

### Expected behaviour
Files should be uploaded/downloaded/links should open in browser

### Actual behaviour
Nothing happens

### Configuration
**Operating system:**
Ubuntu 18.04.4 LTS

**Version of Telegram Desktop:**
2.1 (installed 24/04/2020)

**Installation source (Linux Only)** - the official website / GitHub releases / flatpak / snap / distribution package:
Updated automatically

**Comment by ilya-fedin** [comment]
> Updated automatically

that's not a installation source

**Comment by Vince972** [comment]
Hello,
I confirm. I have the exact same problem since the last update (version 2.1, snap version on Xubuntu 20.04). Before that, I didn't have any issue.

**Comment by ilya-fedin** [comment]
Snap package has been ported to xdg-desktop-portal, you must install xdg-desktop-portal

**Comment by r3medy** [comment]
> Snap package has been ported to xdg-desktop-portal, you must install xdg-desktop-portal

Hi,

I'm having a similar issue. 

### Steps to reproduce
1. Attempt to click any link (new or old) in Telegram Desktop.
2. Nothing happens.

### Expected behaviour
When clicking a link or link preview, the default web browser (in this case Chrome) would open the page. 

### Actual behaviour
Nada.

### Configuration
**Operating system:**
Ubuntu 18.04 LTS x86_64 Linux 5.3.0-46-generic 
GNOME Shell 3.28.4

**Version of Telegram Desktop:**
Version: telegram-desktop 2.1.0 
Install Date: 2020/04/24

**Installation source (Linux Only)** - the official website / GitHub releases / flatpak / snap / distribution package:
Snap package.

**Used theme**:
Default.

I have installed `xdg-desktop-portal` as @ilya-fedin suggested and it hasn't resolved the issue. Are there any other dependencies Snap wouldn't have handled?

**Comment by Vince972** [comment]
> Snap package has been ported to xdg-desktop-portal, you must install xdg-desktop-portal

I have checked. It was already installed.

**Comment by andreyeltsin** [comment]
> Snap package has been ported to xdg-desktop-portal, you must install xdg-desktop-portal

xdg-desktop-portal is already the newest version (1.0.3-0ubuntu0.2).

**Comment by BCommeine** [comment]
I am having the same issue since an update.
Installed using the snap package.
Ubuntu 18.04.4 LTS

**Comment by ilya-fedin** [comment]
> it hasn't resolved the issue

Check that at least one backend is installed (xdg-desktop-portal-gtk, xdg-desktop-portal-kde)

**Comment by outa** [comment]
> > it hasn't resolved the issue
> 
> Check that at least one backend is installed (xdg-desktop-portal-gtk, xdg-desktop-portal-kde)

Same issue here. Telegram installed from snap, `xdg-desktop-portal` version 1.4.2-2ubuntu1 installed; `xdg-desktop-portal-kde` version 5.16.5-0ubuntu2 installed.

**Comment by ilya-fedin** [comment]
> Same issue here.

Does file dialogs work?

**Comment by outa** [comment]
no, neither clicking on a file to download it, nor clicking the clip icon next to the message field does anything.

**Comment by ilya-fedin** [comment]
`ps aux | grep xdg-desktop-portal`

**Comment by outa** [comment]
outa      6253  0.0  0.0   9404   916 pts/1    S+   19:16   0:00 grep --color=auto xdg-desktop-portal

**Comment by ilya-fedin** [comment]
xdg-deskto-portal is not running

**Comment by outa** [comment]
ok, do you have an idea how to (auto-)start it or set it up correctly?

**Comment by ilya-fedin** [comment]
It should be automatically, maybe you forgot to restart system after installation?

**Comment by outa** [comment]
the xdg-desktop-portal packages were already installed, i didn't manually install them. i have also rebooted since telegram updated to v2.1.

**Comment by ilya-fedin** [comment]
`busctl --user tree org.freedesktop.portal.Desktop`

**Comment by outa** [comment]
```
└─/org
  └─/org/freedesktop
    └─/org/freedesktop/portal
      └─/org/freedesktop/portal/desktop
```

**Comment by ilya-fedin** [comment]
`ps aux | grep xdg-desktop-portal`

**Comment by outa** [comment]
```
outa      6906  0.0  0.1 318032 12172 ?        Ssl  19:35   0:00 /usr/libexec/xdg-desktop-portal
outa      6912  0.0  1.0 1043888 77532 ?       Sl   19:35   0:00 /usr/lib/x86_64-linux-gnu/libexec/xdg-desktop-portal-kde
outa      7280  0.0  0.0   9404   844 pts/1    S+   19:42   0:00 grep --color=auto xdg-desktop-portal
```
wow thanks, seems to work now.. also clicking links and uploading works now. clicking a file to download it still not.

but why did it start now? because of the `busctl` command?

**Comment by r3medy** [comment]
Can confirm that fixed my link-click issue too, thanks @ilya-fedin. I've also rebooted since upgrading Telegram and installing `xdg-desktop-portal`, so it definitely didn't start automatically on my machine.

**Comment by ilya-fedin** [comment]
> but why did it start now? because of the `busctl` command?

That's strange. xdg-desktop-portal is dbus-activatable. I.e. when tdesktop tries to call it, it shpuld start automatically, as when you typed busctl command. But for some reason it is not so. Maybe, snap sandbox prevents for starting portal :thinking:

**Comment by r3medy** [comment]
> > but why did it start now? because of the `busctl` command?
> 
> That's strange. xdg-desktop-portal is dbus-activatable. I.e. when tdesktop tries to call it, it shpuld start automatically, as when you typed busctl command. But for some reason it is not so. Maybe, snap sandbox prevents from starting the portal

That would make sense, since you don't need the `--classic` flag when installing via snap.

**Comment by ilya-fedin** [comment]
Anyway, when #7713 will be merged it should fallback to the old behavior if portal is not running

**Comment by a37h** [comment]
I just wanted to mention that I'm having the same issue recently. Clicking links doesn't do anything, downloading files doesn't work. Ubuntu 18.04 and so on. Tag me if you need some more debug info.

**Comment by Qik000** [comment]
Having the same issue, xdg desktop portal is the newest version and running, and i cant follow links, download anything etc.

I installed from SnapCraft

**Comment by ilya-fedin** [comment]
Should be fixed on edge channel

**Comment by Vince972** [comment]
Hello everybody. I don't know if it's a useful information, but you have to know that there is not a such issue with the flatpak version.
That's why I'm using it as a workaround for the moment.

**Comment by ilya-fedin** [comment]
> Should be fixed on edge channel

Any feedback?

**Comment by r3medy** [comment]
> > Should be fixed on edge channel
> 
> Any feedback?

Doesn't seem to have fixed the issue. I now get a pop up confirming "Open Link?" when clicking, but after choosing "Yes", nothing happens. 

As soon as I then run `busctl --user tree org.freedesktop.portal.Desktop` without rebooting or closing telegram, the link opens no problem.

**Comment by Vince972** [comment]
Feed back :
- Ok, links are opening now.
- One thing I have noticed (same on Flatpak version) : when you click on a file that you want to download, the window opens for you to choose you saving location, but there is no file name. It's always empty.
Should I open another issue ?

**Comment by internetcoffeephone** [comment]
> > Should be fixed on edge channel
> 
> Any feedback?

This does not fix the issue for me. No popup asking for open link confirmation either like mentioned [here](https://github.com/telegramdesktop/tdesktop/issues/7701#issuecomment-619994652).

Running Ubuntu 18.04, edge channel installed using `snap refresh telegram-desktop --channel=latest/edge` (just a sanity check, is this the correct channel?).

**Comment by ilya-fedin** [comment]
i use `snap refresh --edge telegram-desktop` usually

**Comment by internetcoffeephone** [comment]
> i use `snap refresh --edge telegram-desktop` usually

That seems to do the same thing.

According to [snapcraft](https://snapcraft.io/telegram-desktop), the stable version is currently the same as the edge version: `2.1.0-5-ge149f10d4`. Is this intentional? I'm assuming it means the edge and stable channel do not differ.

**Comment by ilya-fedin** [comment]
>     * Should I open another issue ?

No, reported here: https://github.com/flatpak/xdg-desktop-portal-gtk/issues/191

**Comment by Aokromes** [comment]
so, 3rd party bug?

**Comment by ilya-fedin** [comment]
> so, 3rd party bug?

This issue - no, issue about @Vince972 talking - yes

**Comment by Darkzarich** [comment]
I noticed similar problem as well. I'm using Ubuntu Ubuntu 18.04.4 LTS. Telegram autoupdated 1-2 days ago and after that for some reason I see pointer cursor when it's on the chats and click on a link doesn't open a new tab in my default browser. This weird behavior wasn't present before said update. Is there any solution to this?

**Comment by ilya-fedin** [comment]
> Is there any solution to this?

#7742

**Comment by Darkzarich** [comment]
Ah, sorry. I didn't notice. thank you!

**Comment by sergeG** [comment]
In my case browser open links but do not open PDF files in chats. Ubuntu 18.04 latest snap from stable channel.

**Comment by sergeG** [comment]
The following warning appears in system journal:

 apparmor="DENIED" operation="dbus_method_call"  bus="session" path="/org/freedesktop/portal/desktop" interface="org.freedesktop.portal.OpenURI" member="OpenFile" mask="send" name="org.freedesktop.portal.Desktop" pid=4987 label="snap.telegram-desktop.telegram-desktop"

**Comment by sergeG** [comment]
Is the issue with file opening should be filed as the separate bug?

**Comment by ilya-fedin** [comment]
> The following warning appears in system journal

Yes, I know

> Is the issue with file opening should be filed as the separate bug?

No

**Comment by ilya-fedin** [comment]
Could anyone test on 2.1.0-37-g9c66bd553?

**Comment by internetcoffeephone** [comment]
> Could anyone test on 2.1.0-37-g9c66bd553?

This version is not available on snapcraft, but the link bug is fixed for me on 2.1.0-41-gae6474748.

**Comment by ilya-fedin** [comment]
Glad to hear!

**Comment by ilya-fedin** [comment]
@Aokromes i think you can close this too

**Comment by r3medy** [comment]
Thanks for all the help @ilya-fedin. Little late to the party but fixed on `telegram-desktop (edge) 2.1.1-1-g5180d31b4`!
